assembly/src/parsers/serde/serialization.rs

```diff
@@ -0,0 +1,466 @@
+use super::{
+    ByteWriter, Instruction, Node, OpCode, Serializable, SerializationError, IF_ELSE_OPCODE,
+    REPEAT_OPCODE, WHILE_OPCODE,
+};
+
+// NODE SERIALIZATION
+// ================================================================================================
+
+impl Serializable for Node {
+    fn write_into(&self, target: &mut ByteWriter) -> Result<(), SerializationError> {
+        match self {
+            Self::Instruction(i) => i.write_into(target),
+            Self::IfElse(if_clause, else_clause) => {
+                target.write_u8(IF_ELSE_OPCODE);
+
+                if_clause.write_into(target)?;
+
+                else_clause.write_into(target)
+            }
+            Self::Repeat(times, nodes) => {
+                target.write_u8(REPEAT_OPCODE);
+
+                // TODO shouldn't this fail if the repeat is expressed with a value greater than
+                // `u16::MAX`?
+                target.write_u16(*times);
```

Comment:
Not introduced by this PR: should we take this opportunity to fail if the user attempts to repeat with a massive value?

The problem of converting without warning or error is that it increases the magic box effect of the parsing.

Reply:
This was actually an obsolete comment as `times` was already guaranteed to be a `u16`.